pkg/client/unversioned/events.go

```diff
@@ -120,7 +120,7 @@ func (e *events) Patch(incompleteEvent *api.Event, data []byte) (*api.Event, err
 func (e *events) List(opts api.ListOptions) (*api.EventList, error) {
 	result := &api.EventList{}
 	err := e.client.Get().
-		NamespaceIfScoped(e.namespace, len(e.namespace) > 0).
+		Namespace(e.namespace).
```

Comment:
When building the URL, if `len(e.namespace)==0`, the namespace won't be shown in the path. That's how all other clients (e.g., the client for Pod) work.